File: clojure/anagram/anagram.clj
(ns anagram (:require [clojure.string :refer [lower-case]]))

(defn anagram? [word candidate]
  (let [w (lower-case word)
        c (lower-case candidate)]
    (and (not= w c)
         (= (sort w) (sort c)))))

(defn anagrams-for [word candidates]
  (filter (partial anagram? word) candidates))

File: elixir/anagram/anagram.exs
defmodule Anagram do
  @doc """
  Returns all candidates that are anagrams of, but not equal to, 'base'.
  """
  @spec match(String.t, [String.t]) :: [String.t]
  def match(base, candidates) do
    base = String.downcase(base)
    sorted_base = sort(base)
    Enum.filter(candidates,
      fn(candidate) ->
        String.downcase(candidate) != base and sort(candidate) == sorted_base
      end)
  end

  defp sort(string) do
    String.downcase(string)
    |> String.to_char_list
    |> Enum.sort
  end
end
